presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -1741,84 +1741,4 @@ private static void checkState(boolean condition)
     }
 
     private UnscaledDecimal128Arithmetic() {}
-
-    public static Slice doubleToLongDecimal(double value, long precision, int scale)
```

Comment:
I am not sure we should give up such easily on optimized conversion. Is there a code we could reuse to implement conversion correctly? What does Hive use?

Reply:
It does not use this implementation (as results are different). I don't know if their implementation uses intermediate `String` object.